pkg/controllers/monitoring-stack/controller.go

```diff
@@ -196,3 +219,34 @@ func (r *reconciler) reconcileObject(ctx context.Context, ms *stack.MonitoringSt
 	logger.Info("Updating stack component")
 	return r.k8sClient.Update(ctx, desired)
 }
+
+func (r *reconciler) createGrafanaDSWatch(ctx context.Context) error {
+	if r.shouldCreateGrafanaDSWatch {
```

Comment:
nit: it may be more readable if exited early .. e..g

```go 
if !r.shouldCreateGrafanaDSWatch {
  return nil
}

// rest of the code to setup the watch ...
```

Reply:
Done